[1.68.0] 45428: Add `supportsMinimalRBACPrivilege` feature (#262)
replicatedhq/replicated-docs: docs/vendor/packaging-rbac.md

By default, the admin console will create a ClusterRole and ClusterRoleBinding with permissions to all namespaces.
This behavior can be controlled by editing the Application custom resource manifest file. For more information about the Application manifest, see [Application](../reference/custom-resource-application) in the _Custom Resources_ section.

> As listed above, an application can require cluster scoped access across all namespaces on all/wildcard k8 objects or to have access limited to its given namespace.

Reviewer: ```suggestion
As listed above, an application can require cluster-scoped access across all namespaces on all/wildcard Kubernetes objects, or it can have access limited to its given namespace.
```

## Namespace-scoped access

An application vendor can limit the role-based access control (RBAC) grants for the admin console to a single namespace by specifying one of the following options in the Application manifest file.

* `requireMinimalRBACPrivileges` - Applicable to existing clusters only. When set to `true`, the app manager creates a Role and RoleBinding, granting the admin console access to all resources in the namespace, but not to any resources outside of the namespace. Use this option if you want all customer installations to use minimal RBAC.

> * `supportMinimalRBACPrivileges` - Applicable to existing clusters only. When set to `true`, the app manager supports creating a namespace-scoped Role and RoleBinding, granting the admin console access to all resources in the namespace, but not to any resources outside of the namespace. Minimal RBAC is not be used by default. It is used only when the `--use-minimal-rbac` flag is passed to the `kots install` command. Use this option if you want a subset of customer installations to use minimal RBAC.

Reviewer: We don't mention the namespace-scoped part for `requireMinimalRBACPrivileges` right above, so I'm just making the text here match. Roles and RoleBindings are always namespace-scoped, so I think it's fine to not mention it. We mention the namespace scoping elsewhere.
```suggestion
* `supportMinimalRBACPrivileges` - Applicable to existing clusters only. When set to `true`, the app manager supports creating a Role and RoleBinding, granting the admin console access to all resources in the namespace, but not to any resources outside of the namespace. Minimal RBAC is not used by default. It is used only when the `--use-minimal-rbac` flag is passed to the `kots install` command. Use this option if you want a subset of customer installations to use minimal RBAC.
```

### Strict Preflight Checks

> Without access to cluster-scoped resources, some preflight checks are not be able to read the resources. These tools continue to function, but return less data. For more information, see [Creating Preflight Checks and Support Bundles](https://docs.replicated.com/vendor/preflight-support-bundle-creating).

Reviewer: Check me on whether that link is correct, but we should use a relative link.
```suggestion
Without access to cluster-scoped resources, some preflight checks are not be able to read the resources. These tools continue to function, but return less data. For more information, see [Creating Preflight Checks and Support Bundles](/vendor/preflight-support-bundle-creating).
```

Author: Fixed. It also needed "../" in front of vendor, not just "/".

## Namespace-scoped access

An application developer can limit the role-based access control (RBAC) grants for the admin console to a single namespace by specifying one of the following options in the Application manifest file.

* `requireMinimalRBACPrivileges` - When this flag is set, the app manager creates a Role and RoleBinding, granting the admin console access to select resources in the namespace, but not outside of the cluster. Setting this flag applies to all customers.

> * `supportMinimalRBACPrivileges` - Applicable to existing clusters only. When set to `true`, the app manager enables a namespace-scoped Role and RoleBinding on a per-customer basis. Therefore, this setting is not triggered by default in the installation. You must also tell the end user to opt in using the `kots install` command with the `-- use-minimal-rbac` flag set to `true`.

Reviewer: ```suggestion
* `supportsMinimalRBACPrivileges` - Applicable to existing clusters only. When set to `true`, the app manager supports creating a Role and RoleBinding, granting the admin console access to all resources in the namespace, but not to any resources outside of the namespace. Minimal RBAC will not be used by default. It will only be used when the `--use-minimal-rbac` flag is passed to the `kots install` command. If you want a subset of customer installations to use minimal RBAC, use this option.
```

Author: Added most of your suggestions, with tweaks for active voice and verb tense.

### Preflight Checks and Support Bundles

Without access to cluster-scoped resources, some preflight checks and support bundle collectors are not be able to read the resources. These tools continue to function, but return less data. In this situation, the admin console presents an option for the user to either proceed with limited data or use the displayed `kubectl preflight` command to run the preflight checks or support bundle remotely with the user's RBAC authorizations. After the command runs, the results are automatically uploaded to the app manager.

**Example: `kubectl preflight` command**

```bash
curl https://krew.sh/preflight | bash
kubectl preflight secret/<namespace>/kotsadm-<appslug>-preflight

> ```

Author: This was copy/pasted (and removed) from the reference topic for Application customs resources that was done with the Blue Team recently re: strict preflights. I'm don't recall why support bundles was included in this language. 

@JustinN-Replicated - Can you help me to clarify the issues that Alex brought up? Please take a look at this topic in the preview for the full context. https://deploy-preview-262--replicated-docs.netlify.app/vendor/packaging-rbac#namespace-scoped-access

> ```

Author: @JustinN-Replicated nevermind. I'm going to put this section to the way we had it before in the Application custom resource topic.

    verb: "*"
```

:::note
The kotsadm-operator component receives an authorization for all verbs, resources, and apiGroups in the namespace.
:::

## Converting

> The RBAC permissions are set during the initial installation. The admin console runs using the assumed identity and cannot change its own authorization. Changing the RBAC scope from cluster to namespace or conversely affects only new installations . Existing installations continue to run with their current authorization.

Reviewer: I take it my use of "vice versa" didn't cut it, but I don't think this reads right with conversely either. Especially since it's followed by "affects", so I read this as "conversely affects". Does this work?
```suggestion
The RBAC permissions are set during the initial installation. The admin console runs using the assumed identity and cannot change its own authorization. Changing the RBAC scope from cluster to namespace or from namespace to cluster affects only new installations. Existing installations continue to run with their current authorization.
```

    verb: "*"
```

:::note
The kotsadm-operator component receives an authorization for all verbs, resources, and apiGroups in the namespace.
:::

## Converting

> The RBAC permissions are set during the initial installation. The admin console is running using the assumed identity and cannot change its own authorization. Changing the RBAC scope from cluster to namespace or from namespace to cluster affects only new installations of the application; existing installations continue to run with their current authorization.

Reviewer: ```suggestion
The RBAC permissions are set during the initial installation. The admin console runs using the assumed identity and cannot change its own authorization. If you change the RBAC scope for your application from cluster to namespace or vice versa, this will only affect new installations. Existing installations will continue to run with their current authorization.
```

Author: Implemented most of the changes. We avoid using Latin words like via and vice-versa because they do not translate well. See what you think about "conversely".
